model/src/payment/payment.rs

```diff
@@ -4,6 +4,17 @@ use bigdecimal::BigDecimal;
 use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Signed<T> {
+    #[serde(flatten)]
+    pub payload: T,
+    #[serde(with = "serde_bytes")]
+    pub signature: Option<Vec<u8>>,
+    #[serde(with = "serde_bytes")]
+    pub signed_bytes: Option<Vec<u8>>,
+}
```

Comment:
It would be better to have single structure which would be optional with non optional fields.
In this code you can have all combinations of Nones, some of which don't make sense.